frontend/src/host_orchestrator/orchestrator/stopcvdaction.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrator
+
+import (
+	"log"
+
+	"github.com/google/android-cuttlefish/frontend/src/host_orchestrator/orchestrator/cvd"
+	apiv1 "github.com/google/android-cuttlefish/frontend/src/liboperator/api/v1"
+	"github.com/google/android-cuttlefish/frontend/src/liboperator/operator"
+)
+
+type StopCVDActionOpts struct {
+	Paths            IMPaths
+	OperationManager OperationManager
+	ExecContext      ExecContext
+	CVDToolsVersion  AndroidBuild
+	CVDDownloader    CVDDownloader
+	CVDUser          string
+}
+
+type StopCVDAction struct {
+	paths           IMPaths
+	om              OperationManager
+	cvdToolsVersion AndroidBuild
+	cvdDownloader   CVDDownloader
+	execContext     cvd.CVDExecContext
+}
+
+func NewStopCVDAction(opts StopCVDActionOpts) *StopCVDAction {
+	return &StopCVDAction{
+		paths:           opts.Paths,
+		om:              opts.OperationManager,
+		cvdToolsVersion: opts.CVDToolsVersion,
+		cvdDownloader:   opts.CVDDownloader,
+		execContext:     newCVDExecContext(opts.ExecContext, opts.CVDUser),
+	}
+}
+
+func (a *StopCVDAction) Run() (apiv1.Operation, error) {
+	op := a.om.New()
+	go func(op apiv1.Operation) {
+		result := &OperationResult{}
+		result.Value, result.Error = a.exec(op)
+		if err := a.om.Complete(op.Name, result); err != nil {
+			log.Printf("error completing operation %q: %v\n", op.Name, err)
+		}
+	}(op)
+	return op, nil
+}
+
+func (a *StopCVDAction) exec(op apiv1.Operation) (*apiv1.StopCVDResponse, error) {
+	if err := a.cvdDownloader.Download(a.cvdToolsVersion, a.paths.CVDBin(), a.paths.FetchCVDBin()); err != nil {
+		return nil, operator.NewInternalError("failed downloading the cvd binary", err)
+	}
+	cmdOpts := cvd.CommandOpts{
+		Home: a.paths.RuntimesRootDir,
+	}
+	cmd := cvd.NewCommand(a.execContext, a.paths.CVDBin(), []string{"stop"}, cmdOpts)
+	if err := cmd.Run(); err != nil {
+		return nil, operator.NewInternalError("cvd stop failed", err)
+	}
```

Comment:
How does this work when you have created two instances in the same host?

Reply:
`cvd stop` stops all the instances, tested this running `cvd directly`. as creating multiple instances in HO is currently broken.

Reply:
This is a deficiency of `cvd` that we can't fix now, but we don't have to carry it over to the HO. Can we use selectors to tell `cvd` what instance we want to stop? It's ok if it stops everything in the group now, but when that is fixed we'll get the correct behavior without any future work needed on our part.

Reply:
I see, will update this PR with selectors, however we need to be able to create multiple groups/cvds  to test `cvd stop` with selectors, creating multiple groups/instances is broken. I'll update the PR after it's fixed.

Reply:
implemented cvd selectors for `cvd stop`. PTAL